### ISSUE TITLE
Avaliar modelos no KinFaceW

### ISSUE THREAD

**Issue by vitalwarley** [vitalwarley/research#88]

**Comment by vitalwarley** [comment]
Adicionei apenas uma operação de finetuning: carrego o modelo treinado no FIW e ajusto no KinFace I e II. 

Resultados que reportei na qualificação

![image](https://github.com/user-attachments/assets/05c03672-d903-4284-b95e-9e8c102e89d2)

_Ours (fine-tuned)_ é o que estou atualmente fazendo. Preciso que meus resultados sejam melhores/competitivos com 0.815 e 0.906 nos datasets I e II, respectivamente.

**Comment by vitalwarley** [comment]
Há várias questões aqui: o que mudar nos hparams? lr? HCL alpha? É bom também investigar se minha implementação está correta. Talvez usar os SOTAs anteriores?

**Comment by vitalwarley** [comment]
Resultados do meu modelo (2024, qualificação, artigo) treinado recentemente (02/2025):

```
Mean Accuracy and AUC by Dataset:
dataset  batch_size      lr  accuracy    auc       fd       fs       md       ms  avg_kinship
      I          16 0.00001   0.64875 0.6745 0.663462 0.641555 0.657500 0.622306     0.646206
      I          16 0.00100   0.71180 0.7662 0.697863 0.717350 0.732462 0.698325     0.711500
     II          16 0.00001   0.77960 0.8430 0.722000 0.806000 0.808000 0.782000     0.779500
     II          16 0.00100   0.86820 0.9162 0.802000 0.874000 0.924000 0.872000     0.868000
```

Na qualificação apresentei os resultados com lr=0.001, então aqui temos 0.71 e 0.86 para os datasets I e II, respectivamente. São resultados equivalentes ao que apresentei. Todavia, longe dos resultados dos trabalhos que comparei.

**Comment by vitalwarley** [comment]
O script que fiz retorna o último valor do experimento, não o melhor. Limitação do guildai. Não encontrei como resolver isso diretamente. Vou avaliar como contornar isso via tensorboard.

Em curso deixei experimentos com _image size_ original -- 64x64. Eu tinha feito para 112x112 achando que ia dar ruim, mas quem sabe.

Outra coisa a fazer dps é experimentar com minha reprodução do baseline (Zhang, 2021), embora com AdaFace em vez de ArcFace.

**Comment by vitalwarley** [comment]
> O script que fiz retorna o último valor do experimento, não o melhor. Limitação do guildai. Não encontrei como resolver isso diretamente. Vou avaliar como contornar isso via tensorboard.

Acho que consegui melhorar algo. Por exemplo, 

```
label dataset  batch_size      lr  accuracy      auc       fd       fs       md       ms  avg_kinship
   v0       I          16 0.00100  0.731886 0.773195 0.697578 0.733562 0.768000 0.725965     0.731276
   v0      II          16 0.00100  0.881000 0.928315 0.808000 0.898000 0.920000 0.898000     0.881000
```

Ainda assim, não é suficiente, especialmente para o dataset I.

**Comment by vitalwarley** [comment]
> Em curso deixei experimentos com _image size_ original -- 64x64. Eu tinha feito para 112x112 achando que ia dar ruim, mas quem sabe.

Na verdade modifiquei o datamodule errado. Não dá pra fazer experimento com 64x64 com os modelos que temos.



> Outra coisa a fazer dps é experimentar com minha reprodução do baseline (Zhang, 2021), embora com AdaFace em vez de ArcFace.

Ainda não fiz. Atualmente estou treinando do zero com ResNet-18 em vez de ResNet-101, como comentado por Su et al (2023), os autores da FaCoRNet

> “Table 4 compares the kinship verification performance by using the pre-trained ResNet-18 in various methods. The result shows that the kinship verification accuracy from our proposed method is comparable to or higher than those achieved by the other methods.” ([Su et al., 2023, p. 6](zotero://select/groups/5590787/items/BUZY9HBF)) ([pdf](zotero://open-pdf/groups/5590787/items/M6SQHBR3?page=6&annotation=9FC64CUT))

**Comment by vitalwarley** [comment]
| label   | weights   | model          |   tau |   alpha_neg | dataset   |   batch_size |     lr |   accuracy |      auc |       fd |       fs |       md |       ms |   avg_kinship |
|:--------|:----------|:---------------|------:|------------:|:----------|-------------:|-------:|-----------:|---------:|---------:|---------:|---------:|---------:|--------------:|
| v0      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 1e-05  |   0.685321 | 0.701507 | 0.675214 | 0.691309 | 0.685385 | 0.685121 |      0.684257 |
| v0      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 0.0001 |   0.75     | 0.784354 | 0.685185 | 0.758065 | 0.78     | 0.782609 |      0.751465 |
| v0      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 0.001  |   0.731886 | 0.773195 | 0.697578 | 0.733562 | 0.768    | 0.725965 |      0.731276 |
| v0      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 1e-05  |   0.7855   | 0.844085 | 0.732    | 0.82     | 0.804    | 0.786    |      0.7855   |
| v0      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 0.001  |   0.881    | 0.928315 | 0.808    | 0.898    | 0.92     | 0.898    |      0.881    |
| v1      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 1e-05  |   0.685321 | 0.701507 | 0.675214 | 0.691309 | 0.685385 | 0.685121 |      0.684257 |
| v1      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 0.0001 |   0.693193 | 0.717685 | 0.682906 | 0.684946 | 0.717385 | 0.675654 |      0.690223 |
| v1      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 0.001  |   0.731886 | 0.773195 | 0.697578 | 0.733562 | 0.768    | 0.725965 |      0.731276 |
| v1      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 1e-05  |   0.7855   | 0.844085 | 0.732    | 0.82     | 0.804    | 0.786    |      0.7855   |
| v1      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 0.0001 |   0.818    | 0.880255 | 0.754    | 0.844    | 0.848    | 0.826    |      0.818    |
| v1      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 0.001  |   0.881    | 0.928315 | 0.808    | 0.898    | 0.92     | 0.898    |      0.881    |
| v3      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 0.001  |   0.708953 | 0.746513 | 0.671225 | 0.708516 | 0.748    | 0.707096 |      0.708709 |
| v3      | e5c261d3  | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 0.001  |   0.879    | 0.92652  | 0.802    | 0.88     | 0.93     | 0.904    |      0.879    |
| v4      |           | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 1e-05  |   0.656884 | 0.671118 | 0.61567  | 0.672997 | 0.646308 | 0.693817 |      0.657198 |
| v4      |           | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 0.0001 |   0.679153 | 0.696072 | 0.652707 | 0.651811 | 0.681385 | 0.676972 |      0.665719 |
| v4      |           | adaface_ir_101 |  0.2  |         0.8 | I         |           16 | 0.001  |   0.702674 | 0.745519 | 0.667806 | 0.693839 | 0.728923 | 0.717504 |      0.702018 |
| v4      |           | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 1e-05  |   0.797    | 0.851095 | 0.738    | 0.824    | 0.818    | 0.808    |      0.797    |
| v4      |           | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 0.0001 |   0.822    | 0.876115 | 0.748    | 0.84     | 0.872    | 0.828    |      0.822    |
| v4      |           | adaface_ir_101 |  0.2  |         0.8 | II        |           16 | 0.001  |   0.879    | 0.923565 | 0.786    | 0.91     | 0.928    | 0.892    |      0.879    |
| v5      |           | adaface_ir_18  |  0.2  |         0.8 | I         |           16 | 0.001  |   0.749756 | 0.805721 | 0.682764 | 0.772542 | 0.795231 | 0.739168 |      0.747426 |
| v5      |           | adaface_ir_18  |  0.2  |         0.8 | II        |           16 | 0.001  |   0.8745   | 0.92395  | 0.78     | 0.91     | 0.912    | 0.896    |      0.8745   |
| v6      |           | adaface_ir_18  |  0.08 |         0   | I         |           16 | 0.001  |   0.747845 | 0.80097  | 0.686325 | 0.748022 | 0.803692 | 0.750894 |      0.747233 |
| v6      |           | adaface_ir_18  |  0.08 |         0   | II        |           16 | 0.001  |   0.874    | 0.929225 | 0.774    | 0.906    | 0.91     | 0.906    |      0.874    |
| v7      | cf554e90  | adaface_ir_18  |  0.2  |         0.8 | I         |           16 | 0.001  |   0.774741 | 0.826864 | 0.731054 | 0.770506 | 0.826923 | 0.780388 |      0.777218 |
| v7      | cf554e90  | adaface_ir_18  |  0.2  |         0.8 | II        |           16 | 0.001  |   0.8805   | 0.93277  | 0.796    | 0.906    | 0.914    | 0.906    |      0.8805   |
| v8      |           | adaface_ir_18  |  0.2  |         0.8 | I         |           16 | 0.001  |   0.749756 | 0.805721 | 0.682764 | 0.772542 | 0.795231 | 0.739168 |      0.747426 |
| v8      |           | adaface_ir_18  |  0.2  |         0.8 | II        |           16 | 0.001  |   0.8745   | 0.92395  | 0.78     | 0.91     | 0.912    | 0.896    |      0.8745   |

Experimentos que fiz com KinFaceWI e II. Ainda não consegui chegar ao que os autores reportaram e estou sem ideias do que fazer.

- v0 - finetuning com meu SOTA (2024)
- v1 - v0 -- achei que tinha removido resize.
- v3 - adicionei normalização ImageNet -- única vez.
- v4 - treinamento do zero com R101.
- v5 - treinamento com R18. 
- v6 - baseline (sem HCL) com R18.
- v7 - finetuning com meu SOTA (mas usando R18).
- v8 - treinamento do zero com R18.

**Comment by vitalwarley** [comment]
Acredito que vou tentar o seguinte:

1. Finetuning FIW com ArcFace à lá Zhang 2021
2. Finetuning KinFaceW com resultado de 1

O esperado é termos piora nos resultados relativo ao v0, considerando a hipótese de que HCL traz melhores resultados do que CL original. Todavia, talvez seja necessário tunar alpha e tau aqui.

Antes disso, vou experimentar um v0 sem HCL.

**Comment by vitalwarley** [comment]
| label   | weights   | model          |   tau |   alpha_neg | dataset   |   batch_size |    lr |   accuracy |      auc |       fd |       fs |       md |       ms |   avg_kinship |
|:--------|:----------|:---------------|------:|------------:|:----------|-------------:|------:|-----------:|---------:|---------:|---------:|---------:|---------:|--------------:|
| v9      | e5c261d3  | adaface_ir_101 |  0.08 |           0 | I         |           16 | 0.001 |   0.746144 | 0.797966 | 0.697293 | 0.752447 | 0.759846 | 0.779541 |      0.747282 |
| v9      | e5c261d3  | adaface_ir_101 |  0.08 |           0 | II        |           16 | 0.001 |   0.895    | 0.949745 | 0.818    | 0.92     | 0.936    | 0.906    |      0.895    |

73% -> 74% para o I e 88% -> 89% para o II.

**Comment by vitalwarley** [comment]
| label   | weights   | model          |   tau |   alpha_neg | dataset   |   batch_size |     lr |   accuracy |      auc |       fd |       fs |       md |       ms |   avg_kinship |
|:--------|:----------|:---------------|------:|------------:|:----------|-------------:|-------:|-----------:|---------:|---------:|---------:|---------:|---------:|--------------:|
| v10     | ea1f097c  | arcface_ir_101 |  0.08 |         0   | I         |           16 | 1e-05  |   0.64009  | 0.65597  | 0.607692 | 0.635531 | 0.665692 | 0.652654 |      0.640392 |
| v10     | ea1f097c  | arcface_ir_101 |  0.08 |         0   | I         |           16 | 0.0001 |   0.691502 | 0.72167  | 0.686325 | 0.681708 | 0.693846 | 0.704066 |      0.691486 |
| v10     | ea1f097c  | arcface_ir_101 |  0.08 |         0   | I         |           16 | 0.001  |   0.763263 | 0.811791 | 0.746011 | 0.74804  | 0.795692 | 0.769132 |      0.764719 |
| v10     | ea1f097c  | arcface_ir_101 |  0.08 |         0   | II        |           16 | 1e-05  |   0.727    | 0.774405 | 0.662    | 0.74     | 0.75     | 0.756    |      0.727    |
| v10     | ea1f097c  | arcface_ir_101 |  0.08 |         0   | II        |           16 | 0.0001 |   0.8055   | 0.863108 | 0.744    | 0.822    | 0.834    | 0.822    |      0.8055   |
| v10     | ea1f097c  | arcface_ir_101 |  0.08 |         0   | II        |           16 | 0.001  |   0.913    | 0.962365 | 0.828    | 0.936    | 0.946    | 0.942    |      0.913    |
| v11     |           | arcface_ir_101 |  0.08 |         0   | I         |           16 | 0.001  |   0.751935 | 0.791753 | 0.716239 | 0.741928 | 0.775846 | 0.776746 |      0.75269  |
| v11     |           | arcface_ir_101 |  0.08 |         0   | II        |           16 | 0.001  |   0.9145   | 0.963365 | 0.83     | 0.946    | 0.946    | 0.936    |      0.9145   |
| v11     |           | arcface_ir_101 |  0.08 |         0   | I         |           16 | 1e-05  |   0.662636 | 0.682321 | 0.619373 | 0.642083 | 0.705385 | 0.682562 |      0.662351 |
| v11     |           | arcface_ir_101 |  0.08 |         0   | I         |           16 | 0.0001 |   0.69043  | 0.716677 | 0.668091 | 0.687579 | 0.713077 | 0.685357 |      0.688526 |
| v11     |           | arcface_ir_101 |  0.08 |         0   | II        |           16 | 1e-05  |   0.723    | 0.760828 | 0.668    | 0.742    | 0.748    | 0.734    |      0.723    |
| v11     |           | arcface_ir_101 |  0.08 |         0   | II        |           16 | 0.0001 |   0.789    | 0.847845 | 0.706    | 0.81     | 0.848    | 0.792    |      0.789    |

- v10 ArcFace FT com 0.7647 @ lr=0.001 no I e 0.913 @ lr=0.001 no II
- v11 ArcFace no-FT com 0.7527 @ lr=0.001 no I e 0.9145 @ lr=0.001 no II

Diferente do que foi [reportado](https://github.com/vitalwarley/research/issues/88#issuecomment-2575803348) por Su et al. E agora? Experimentar com diferentes seeds e obter uma média e desvio padrão também?

**Comment by vitalwarley** [comment]
Estou avaliando ArcFace com HCL (novo experimento) no KinFaceW. Depos vou mais uma vez reproduzir FaCoRNet.

**Comment by vitalwarley** [comment]
- v12 - KinFace FT (no-HCL) com ArcFace + HCL  com 0.7556 @ lr=0.001 no I e 0.917 @ lr=0.001 no II
  - Treinei ArcFace + HCL no FIW e usei o modelo para treino no KinFace (mas sem HCL). 
- v13 - KinFace FT (HCL) com ArcFace + HCL  com 0.7570 @ lr=0.001 no I e 0.9025 @ lr=0.001 no II

| label   | weights   | model          |   tau |   alpha_neg | dataset   |   batch_size |    lr |   accuracy |      auc |       fd |       fs |       md |       ms |   avg_kinship |
|:--------|:----------|:---------------|------:|------------:|:----------|-------------:|------:|-----------:|---------:|---------:|---------:|---------:|---------:|--------------:|
| v12     | 94bc4893  | arcface_ir_101 |  0.08 |           0 | I         |           16 | 0.001 |   0.755616 | 0.812845 | 0.694017 | 0.759049 | 0.772154 | 0.805232 |      0.757613 |
| v12     | 94bc4893  | arcface_ir_101 |  0.08 |           0 | II        |           16 | 0.001 |   0.917    | 0.962295 | 0.864    | 0.928    | 0.942    | 0.934    |      0.917    |
| v13     | 94bc4893  | arcface_ir_101 |   0.2 |         0.8 | I         |           16 | 0.001 |   0.756842 | 0.800859 | 0.698291 | 0.751101 | 0.815231 | 0.763702 |      0.757081 |
| v13     | 94bc4893  | arcface_ir_101 |   0.2 |         0.8 | II        |           16 | 0.001 |   0.9025   | 0.949155 | 0.85     | 0.902    | 0.926    | 0.932    |      0.9025   |

**Comment by vitalwarley** [comment]
> Depos vou mais uma vez reproduzir FaCoRNet (v14).

| label   | weights   | model          |   s | dataset   |   batch_size |     lr |   accuracy |      auc |       fd |       fs |       md |       ms |   avg_kinship |
|:--------|:----------|:---------------|----:|:----------|-------------:|-------:|-----------:|---------:|---------:|---------:|---------:|---------:|--------------:|
| v14     | 8efcaf16  | adaface_ir_101 | 500 | I         |           16 | 1e-05  |   0.680349 | 0.707764 | 0.634473 | 0.675036 | 0.717231 | 0.698165 |      0.681226 |
| v14     | 8efcaf16  | adaface_ir_101 | 500 | I         |           16 | 0.0001 |   0.706556 | 0.747178 | 0.683048 | 0.680002 | 0.741385 | 0.721852 |      0.706572 |
| v14     | 8efcaf16  | adaface_ir_101 | 500 | I         |           16 | 0.001  |   0.747729 | 0.799769 | 0.697436 | 0.759137 | 0.772308 | 0.767655 |      0.749134 |
| v14     | 8efcaf16  | adaface_ir_101 | 500 | II        |           16 | 1e-05  |   0.801    | 0.868885 | 0.73     | 0.826    | 0.826    | 0.822    |      0.801    |
| v14     | 8efcaf16  | adaface_ir_101 | 500 | II        |           16 | 0.0001 |   0.836    | 0.89614  | 0.766    | 0.852    | 0.862    | 0.864    |      0.836    |
| v14     | 8efcaf16  | adaface_ir_101 | 500 | II        |           16 | 0.001  |   0.9045   | 0.95798  | 0.828    | 0.922    | 0.95     | 0.918    |      0.9045   |
